### DISCUSSION TITLE
A Year in BusTub

### DISCUSSION THREAD

**Discussion by skyzh** [skyzh/skyzh-site#25] [General]
Link to the blog post: https://www.skyzh.dev/blog/2023-08-11-a-year-in-bustub/

**Comment by lmwnshn** [comment]
> Wan spent too much time reviewing my PRs that Andy asked him to focus on his research.

I do remember feeling DoS'd by PRs early in the semester :P That said, from reviewing your first few large PRs (binder, and was it the planner?), it was clear that you knew what you were doing and that you wrote good code. After that, we were very comfortable with you having free rein of the codebase.

Personally, what I found even more impressive was your independence and your ability to plan project timelines. I think you hit basically all of your estimates and milestones, and every week's team lunch was another pleasant surprise about new BusTub capabilities. I think the most surprising PR was wasm for web shell, that PR notification came out of nowhere and I dropped whatever I was doing at the time to check it out.

You didn't mention that some students wanted your autograph at the end of the semester! :) And thanks for joining many of my office hours.

**Comment by Lupinus** [comment]
I greatly enjoy learning Bustub as a beginner who has recently entered the field of storage with a keen interest. I am very excited about the content related to LSM and MVCC, hoping to master a 'small but complete' database through studying Bustub.
